File: LLM-Translations/GPT-4/avatar/Python/Go/atcoder_ABC166_C.go
package main

import (
	"fmt"
)

func main() {
	var n, m int
	fmt.Scan(&n, &m)
	p := make([]int, n)
	h := make([]int, n)
	hp := make(map[int]int)
	path := make(map[int][]int)
	for i := 0; i < n; i++ {
		p[i] = i + 1
	}
	for i := 0; i < n; i++ {
		fmt.Scan(&h[i])
		hp[p[i]] = h[i]
		path[i+1] = []int{}
	}
	var a, b int
	for i := 0; i < m; i++ {
		fmt.Scan(&a, &b)
		path[a] = append(path[a], b)
		path[b] = append(path[b], a)
	}
	c := 0
	for i, v := range path {
		f := 0
		for _, j := range v {
			if hp[i] <= hp[j] {
				f = 1
				break
			}
		}
		if f == 0 {
			c++
		}
	}
	fmt.Println(c)
}

// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Java/atcoder_ABC166_C.java

import java.util.*;

public class atcoder_ABC166_C {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = in.nextInt();
        int[] p = new int[n + 1];
        int[] h = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            p[i] = i;
            h[i] = in.nextInt();
        }
        HashMap<Integer, ArrayList<Integer>> path = new HashMap<>();
        for (int i = 1; i <= n; i++) {
            path.put(i, new ArrayList<>());
        }
        for (int i = 0; i < m; i++) {
            int a = in.nextInt();
            int b = in.nextInt();
            path.get(a).add(b);
            path.get(b).add(a);
        }
        int c = 0;
        for (Map.Entry<Integer, ArrayList<Integer>> entry : path.entrySet()) {
            int i = entry.getKey();
            ArrayList<Integer> v = entry.getValue();
            int f = 0;
            for (int j : v) {
                if (h[i] <= h[j]) {
                    f = 1;
                    break;
                }
            }
            if (f == 0) {
                c++;
            }
        }
        System.out.println(c);
    }
}

// End of Code.
